src/main/java/seedu/address/logic/parser/tags/FindTagCommandParser.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.logic.parser.tags;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SUPERTAG_ONLY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Optional;
+
+import seedu.address.commons.core.booleaninput.BooleanInput;
+import seedu.address.commons.util.StringUtil;
+import seedu.address.logic.commands.tags.FindTagCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.NameContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindTagCommand object.
+ */
+public class FindTagCommandParser implements Parser<FindTagCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of an FindTagCommand
+     * and returns it for execution. Either filters by tags that partially match the
+     * keyword, type of tag (super or regular), or both, depending on user input.
+     * @throws ParseException if the user input does not conform to the expected format
+     */
+    public FindTagCommand parse(String args) throws ParseException {
+        assert args != null;
+
+        ArgumentMultimap argumentMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_TAG, PREFIX_SUPERTAG_ONLY);
+
+        Optional<NameContainsKeywordsPredicate> findPredicate = Optional.empty();
```

Comment:
Is there an exception being thrown for st/ values of more than 1 or less than 0?

Reply:
yep, this is handled in the `BooleanInput` class.

![Screenshot 2020-11-07 at 3 16 46 PM](https://user-images.githubusercontent.com/41122695/98434607-4243dd00-210c-11eb-9505-51355055fc3a.png)